### PR TITLE
Remove legacy database types

### PR DESCRIPTION
**Pull request by swansontec**
We keep the legacy routes around, but turn them into no-ops. Registering a new device still works, since we still want to send security notifications to legacy devices.

### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [x] Yes
- [ ] No

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208918820752819

### CODE REVIEW

File: src/server/routes/legacyRoutes.ts [hunk 1]
```diff
@@ -92,23 +102,12 @@ export const fetchStateV1Route: Serverlet<ApiRequest> = async request => {
...
-  const { connection, date, log, query } = request
+  const { connection, date, query } = request
 
   const checkedQuery = checkPayload(asAttachUserQuery, query)
   if (checkedQuery.error != null) return checkedQuery.error
   const { deviceId, userId } = checkedQuery.clean
 
-  const device = await Device.fetch(deviceId)
-  if (device == null) {
-    return errorResponse(`Cannot find device ${deviceId}`, { status: 404 })
```
Comment (samholmes) [inline src/server/routes/legacyRoutes.ts:103]: will the v2 update throw if deviceId isn't found in the new models?

File: src/server/routes/legacyRoutes.ts [hunk 2]
```diff
@@ -161,25 +163,20 @@ export const registerCurrenciesV1Route: Serverlet<
...
 
   const checkedQuery = checkPayload(asUserIdQuery, query)
   if (checkedQuery.error != null) return checkedQuery.error
-  const { userId } = checkedQuery.clean
+  // const { userId } = checkedQuery.clean
 
-  const match = path.match(/notifications\/([0-9A-Za-z]+)\/?$/)
-  const currencyCode = match != null ? match[1] : ''
+  // const match = path.match(/notifications\/([0-9A-Za-z]+)\/?$/)
+  // const currencyCode = match != null ? match[1] : ''
```
Comment (swansontec) [inline src/server/routes/legacyRoutes.ts:173]: I left code like this commented out, so we sill have a reference for what the payload looks like, even if we ignore it.

**Review by samholmes — APPROVED** [review]
This looks like it does what we discussed in our conversation. Just one question.

File: src/server/routes/legacyRoutes.ts (hunk 1 above)
Comment (swansontec) [reply src/server/routes/legacyRoutes.ts:103]: The v2 `getDeviceById` will return an empty fallback device if the lookup fails, so no, it won't cause a problem.

**Pull request merged**